packages/gateway/src/features/registration/root-resolvers.ts

```diff
@@ -636,20 +639,79 @@ export const resolvers: GQLResolver = {
 
       return taskEntry.resource.id
     },
-    async confirmRegistration(_, { id }, { headers: authHeader }) {
+    async confirmRegistration(_, { id, details }, { headers: authHeader }) {     
+      const token = authHeader.Authorization.replace("Bearer ", "").trim();
+      const recordSpecificToken = await tokenExchangeHandler(
+        token,
+        authHeader,
+        id
+      )
```

Comment:
The confirmRegistration & rejectRegistration will be called with the record specific token - the token will already be in the headers. You don't need to fetch it anymore here